### PR TITLE
Adds the Mad Scientist Megaphone (& megaphone refactor)

### PR DESCRIPTION
**Pull request by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE**
[content][wiki]
dreamseeker_2019-05-26_23-44-25

Costs 1 TC, free for any science position
Can broadcast your BIG SPEAK demands and threats into any radio frequency on the station (but you can't hear into it).
Can also hide your voice, if you want.
For letting everyone know you have a big dick and are not afraid to use it.

Still using the base megaphone sprite megaphone if anyone wants to sprite a funky cartoon villain megaphone feel free to

:cl:
 * rscadd: Added the Mad Scientist Megaphone, purchasable for 1 TC. It can mask your voice and transmit (but not receive) into any radio channel, so you can let your demands be known.
 * tweak: Megaphones now work passively as you speak rather than have a textbox.

### CODE REVIEW

**Comment by GlassEclipse** [comment]
Make it more expensive, like 6 TC maybe? It's like the balloon - you gotta let them know that you don't even need TC to fuck over the station.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Making it cheap was specifically the point to incentivize antag gimmicks

**Comment by Lunstra** [comment]
Incredible

**Comment by Cockswain** [comment]
If you make it free for science they can just get an infinite amount... Why not 2 halved to 1 for scientists?

**Comment by tuypo1** [comment]
Not a fan of making microphones tg style passive but the mad scientist microphone itself is pretty cool

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
@Cockswain I plan to repeat the same mistakes and TRUST that it being free won't result in scientists buying 500 of them as a meme and then when they do I will be disappointed and make it 1TC for everyone with a bitchy passive-aggressive changelog saying "this is why we can't have nice things"
@tuypo1 I agree if only because it means people can now megaphone directly into station bounced radios. If enough people bitch I'll disable that somehow

**Comment by Kurfursten** [comment]
if you set the num_in_stock to, say, 20 then worst case a scientist can only shit out 20 of them and it's unlikely that it would affect most shifts

**Comment by mtpowell** [comment]
>Code Rainbow
>20 clowns with 20 megaphones

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
As funny as the mental image of 20 Syndicate-backed assistants shitposting with big letters into every radio is, I personally don't approve because I think radio spam is the bad kind of gunk
So now there are only 3 megaphones in stock
More than 2 traitors doing a megaphone-into-secchannel gimmick per round is probably too much in any case

**Comment by SonixApache** [comment]
>free griff tool for all syndie scientists

*why*

apart from that cute stuff, big 👍

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
If they get it for free MAYBE they'll try doing a bomb threat rather than just 12:10 telebomb

Right?

File: code/datums/uplink_item.dm [hunk 1]
```diff
@@ -44,7 +44,7 @@ var/list/uplink_items = list()
 	var/only_on_month	//two-digit month as string
 	var/only_on_day		//two-digit day as string
 	var/num_in_stock = 0	// Number of times this can be bought, globally. 0 is infinite
-	var/static/times_bought = 0
```
Comment (ShiftyRail) [inline code/datums/uplink_item.dm:47]: Isn't it used for the Chef DO NOT TIP which HAS to be unique ?

Comment (I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE) [reply code/datums/uplink_item.dm]: `/static/ `means ALL uplink thingies use the same number
Which meant buying 1 megaphone caused you to not be able to purchase the do not tip thingy and that purchasing the latter meant you could only buy 2 megaphones
So basically shitcode
I tested it after my fix, and tested spawning a second uplink, seems to work

Comment (ShiftyRail) [reply code/datums/uplink_item.dm:47]: Nice
I trust you to have it tested

**Comment by SonixApache** [comment]
I still think FREE is silly.

**Comment by Allspice** [comment]
if people bitch hard enough it could be made to only make the first one free

**Comment by Blithering** [comment]
This will definitely be used for reading porn over sec channel. Encouraging gimmicks is a maybe though.

**Pull request merged**